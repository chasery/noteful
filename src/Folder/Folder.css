.Folder {
  list-style: none;
  border: 0.0625rem solid #0000001c;
}

.Folder .active {
  background-color: lightblue;
}

.Folder + .Folder {
  border-top: 0;
}

.Folder:first-child {
  border-radius: 0.25rem 0.25rem 0 0;
}

.Folder:last-child {
  border-radius: 0 0 0.25rem 0.25rem;
}

.Folder__link {
  display: block;
  padding: 1rem 0.5rem;
}

.Folder__title {
  margin: 0;
  font-size: 1.25rem;
  text-align: center;
}
