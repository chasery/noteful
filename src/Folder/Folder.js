import React, { Component } from "react";
import PropTypes from "prop-types";
import { NavLink } from "react-router-dom";
import "./Folder.css";

class Folder extends Component {
  render() {
    return (
      <li className="Folder">
        <NavLink className="Folder__link" to={`/folder/${this.props.id}`}>
          <h2 className="Folder__title">{this.props.name}</h2>
        </NavLink>
      </li>
    );
  }
}

Folder.defaultProps = {
  id: 1,
  name: "Folder",
};

Folder.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
};

export default Folder;
