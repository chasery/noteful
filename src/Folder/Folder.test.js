import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Folder from "./Folder";

describe("Folder component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<Folder />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
