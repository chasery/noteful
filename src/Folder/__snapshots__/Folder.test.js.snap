// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Folder component renders without error 1`] = `
<li
  className="Folder"
>
  <NavLink
    className="Folder__link"
    to="/folder/1"
  >
    <h2
      className="Folder__title"
    >
      Folder
    </h2>
  </NavLink>
</li>
`;
